Support for treatment strategies with interventions at non-consecutiv…

jmafoster1 opened the pull request:
Support for treatment strategies with interventions at non-consecutive timesteps. Timesteps without an explicitly specified intervention are treated as "don't care", so anything can happen.

jmafoster1 commented:
@f-allian, is the codecov just failing on this because the codebase has got slightly bigger? It says I have 100% coverage of the patch, so that's all I can think of, but adding in extra code with 100% coverage should be fine right?

jmafoster1 commented on causal_testing/specification/capabilities.py:
Yes, this was intentional. It is no longer required as the new functionality requires timesteps to be specified explicitly, which is what I wrote `capabilities.py` to handle implicitly.

f-allian reviewed (approved):
@jmafoster1 Feel free to merge following our meeting today :)

jmafoster1 commented:
Pretty sure I've fixed all the bugs in this. @f-allian , I've re-requested a review as I've changed quite a lot of stuff. Most of it is quite technical, but please let me know if anything seems odd.

f-allian reviewed (approved):
@jmafoster1 The new changes all look good to me
